tools/trap/trap.c

```diff
@@ -24,6 +24,14 @@
  ****************************************************************************/
 
 #include "trap.h"
+#include "../../os/include/tinyara/config.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#ifndef CONFIG_SMARTFS_MAXNAMLEN
+#define USERFS_MAXNAMLEN 32
+#endif
```

Comment:
Even you include the header, is this necessary?
If you think some config which does not have the config in the .config, it could be fs disabled.

Reply:
In case fs is not disabled
The application will return an error while looking for the file that the user has asked to dump.

Also, I believe that the user may be using userfs with SmartFS enabled but may not build the code themselves and may not have the .config file available.
Hence, there are other ways to provide the parameter of maximum file name length.